### ISSUE TITLE
There is always an osd process that takes up high cpu and memory.

### ISSUE THREAD

**Issue by cyxinda** [rook/rook#13901]
* Bug Report
There is always an osd process that takes up high cpu at a certain moment for about 4 months .
![1111](https://github.com/rook/rook/assets/7237898/9e4e0302-16b9-4d55-aca5-88bfbf869eab)
I found that this is an admin_socket process, which belongs to the osd pod(pid:2809092) .
![2222](https://github.com/rook/rook/assets/7237898/db30263a-2f15-4ddc-b531-396307b1ed87)
![3333](https://github.com/rook/rook/assets/7237898/4fd018d6-88ab-4544-b73e-b2473f51c98c)
```bash
[root@sc-node-ceph-4 ~]# pstree -apscl 3360113
systemd,1 --switched-root --system --deserialize 31
  └─containerd-shim,1229280 -namespace k8s.io -id 712f8293ab1ecd3d5cc0e576efad6ed9f4e943ccabf14834fd98241e3363a988 -address /run/containerd/containerd.sock
      └─ceph-osd,2809092 --foreground --id 29 --fsid aa0e7bed-d3e3-49a7-a471-8e354bfe61f6 --setuser ceph --setgroup ceph --crush-location=root=default host=sc-node-ceph-4 --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true --default-log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false
          └─admin_socket,3360113 --foreground --id 29 --fsid aa0e7bed-d3e3-49a7-a471-8e354bfe61f6 --setuser ceph --setgroup ceph --crush-location=root=default host=sc-node-ceph-4 --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true --default-log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false
```
After the above osd pod is restarted, another osd pod will replace it at another moment and occupy high CPU. This phenomenon occurs again and again.
![444](https://github.com/rook/rook/assets/7237898/9dca1076-5712-4d47-bbb2-86493f797cb8)
I found that the admin_socket process belonging to a same osd pod(pid:2809092) is also constantly restarting.
```bash
[root@sc-node-ceph-4 ~]# pstree -apscl 3364029
systemd,1 --switched-root --system --deserialize 31
  └─containerd-shim,1229280 -namespace k8s.io -id 712f8293ab1ecd3d5cc0e576efad6ed9f4e943ccabf14834fd98241e3363a988 -address /run/containerd/containerd.sock
      └─ceph-osd,2809092 --foreground --id 29 --fsid aa0e7bed-d3e3-49a7-a471-8e354bfe61f6 --setuser ceph --setgroup ceph --crush-location=root=default host=sc-node-ceph-4 --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true --default-log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false
          └─admin_socket,3364029 --foreground --id 29 --fsid aa0e7bed-d3e3-49a7-a471-8e354bfe61f6 --setuser ceph --setgroup ceph --crush-location=root=default host=sc-node-ceph-4 --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true --default-log-stderr-prefix=debug  --default-log-to-file=false --default-mon-cluster-log-to-file=false
```
The status of ceph cluster is HEALTH.
```bash
 ✘ ⚡ root@sc-master-1  ~/rooks  kubectl -n rook-ceph exec -it rook-ceph-tools-7d4b5bb689-k5tvp -- /bin/bash
bash-4.4$ ceph -s
  cluster:
    id:     aa0e7bed-d3e3-49a7-a471-8e354bfe61f6
    health: HEALTH_OK
 
  services:
    mon: 3 daemons, quorum a,e,j (age 7h)
    mgr: b(active, since 6w), standbys: a
    mds: 1/1 daemons up, 1 hot standby
    osd: 36 osds: 36 up (since 7h), 36 in (since 3M)
    rgw: 3 daemons active (3 hosts, 1 zones)
 
  data:
    volumes: 1/1 healthy
    pools:   12 pools, 649 pgs
    objects: 37.40M objects, 16 TiB
    usage:   47 TiB used, 84 TiB / 131 TiB avail
    pgs:     647 active+clean
             2   active+clean+scrubbing+deep
 
  io:
    client:   7.9 KiB/s rd, 1.9 MiB/s wr, 3 op/s rd, 137 op/s wr
```
**Deviation from expected behavior:**
There is always an osd process that takes up high cpu and memory .And these osd processes appear alternately with restarts.
**Expected behavior:**
This strange admin-socket process should not occur and should not cause the osd pod to keep restarting
  
 **The osd pod part logs**
```bash
[root@sc-node-ceph-4 ~]# nerdctl -n k8s.io logs -f 6897f338b6ce
rotating pattern: /var/log/ceph/ceph-osd.29.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
considering log /var/log/ceph/ceph-osd.29.log
  Now: 2024-03-08 10:30
  Last rotated at 2024-03-08 00:00
  log does not need rotating (log has been rotated at 2024-3-8 0:0, that is not day ago yet)
not running postrotate script, since no logs were rotated
switching euid to 0 and egid to 0
+ sleep 15m
+ true
+ logrotate --verbose /etc/logrotate.d/ceph
reading config file /etc/logrotate.d/ceph
Reading state from file: /var/lib/logrotate/logrotate.status
Allocating hash table for state file, size 64 entries
Creating new state

Handling 1 logs

rotating pattern: /var/log/ceph/ceph-osd.29.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
considering log /var/log/ceph/ceph-osd.29.log
  Now: 2024-03-08 10:45
  Last rotated at 2024-03-08 00:00
  log does not need rotating (log has been rotated at 2024-3-8 0:0, that is not day ago yet)
not running postrotate script, since no logs were rotated
switching euid to 0 and egid to 0
+ sleep 15m
+ true
+ logrotate --verbose /etc/logrotate.d/ceph
reading config file /etc/logrotate.d/ceph
Reading state from file: /var/lib/logrotate/logrotate.status
Allocating hash table for state file, size 64 entries
Creating new state

Handling 1 logs

rotating pattern: /var/log/ceph/ceph-osd.29.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
considering log /var/log/ceph/ceph-osd.29.log
  Now: 2024-03-08 11:00
  Last rotated at 2024-03-08 00:00
  log does not need rotating (log has been rotated at 2024-3-8 0:0, that is not day ago yet)
not running postrotate script, since no logs were rotated
switching euid to 0 and egid to 0
+ sleep 15m
+ true
+ logrotate --verbose /etc/logrotate.d/ceph
reading config file /etc/logrotate.d/ceph
Reading state from file: /var/lib/logrotate/logrotate.status
Allocating hash table for state file, size 64 entries
Creating new state

Handling 1 logs

rotating pattern: /var/log/ceph/ceph-osd.29.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
considering log /var/log/ceph/ceph-osd.29.log
  Now: 2024-03-08 11:15
  Last rotated at 2024-03-08 00:00
  log does not need rotating (log has been rotated at 2024-3-8 0:0, that is not day ago yet)
not running postrotate script, since no logs were rotated
switching euid to 0 and egid to 0
+ sleep 15m
+ true
+ logrotate --verbose /etc/logrotate.d/ceph
reading config file /etc/logrotate.d/ceph
Reading state from file: /var/lib/logrotate/logrotate.status
Allocating hash table for state file, size 64 entries
Creating new state

Handling 1 logs

rotating pattern: /var/log/ceph/ceph-osd.29.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
considering log /var/log/ceph/ceph-osd.29.log
  Now: 2024-03-08 11:30
  Last rotated at 2024-03-08 00:00
  log does not need rotating (log has been rotated at 2024-3-8 0:0, that is not day ago yet)
not running postrotate script, since no logs were rotated
switching euid to 0 and egid to 0
+ sleep 15m
+ true
+ logrotate --verbose /etc/logrotate.d/ceph
reading config file /etc/logrotate.d/ceph
Reading state from file: /var/lib/logrotate/logrotate.status
Allocating hash table for state file, size 64 entries
Creating new state

Handling 1 logs

rotating pattern: /var/log/ceph/ceph-osd.29.log  after 1 days (7 rotations)
empty log files are not rotated, log files >= 524288000 are rotated earlier, old logs are removed
switching euid to 0 and egid to 167
considering log /var/log/ceph/ceph-osd.29.log
  Now: 2024-03-08 11:45
  Last rotated at 2024-03-08 00:00
  log does not need rotating (log has been rotated at 2024-3-8 0:0, that is not day ago yet)
not running postrotate script, since no logs were rotated
switching euid to 0 and egid to 0
+ sleep 15m
```

**Cluster Status to submit**:
 
**Environment**:
* OS (e.g. from /etc/os-release):
```bash
 ⚡ root@sc-master-1  ~/rooks/rook-1.12.10/deploy/examples  cat /etc/os-release      
NAME="Rocky Linux"
VERSION="9.2 (Blue Onyx)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.2"
PLATFORM_ID="platform:el9"
PRETTY_NAME="Rocky Linux 9.2 (Blue Onyx)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:9::baseos"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2032-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-9"
ROCKY_SUPPORT_PRODUCT_VERSION="9.2"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.2"
```
* Kernel (e.g. `uname -a`):
```bash
 ⚡ root@sc-master-1  ~/rooks/rook-1.12.10/deploy/examples  uname -a                  
Linux sc-master-1 6.6.2-1.el9.elrepo.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Nov 20 12:18:26 EST 2023 x86_64 x86_64 x86_64 GNU/Linux
```
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): V1.12.10
* Storage backend version (e.g. for ceph do `ceph -v`):
```bash
bash-4.4$ ceph -v
ceph version 17.2.6 (d7ff0d10654d2280e08f1ab989c7cdf3064446a5) quincy (stable)
```
* Kubernetes version (use `kubectl version`):
```bash
 ⚡ root@sc-master-1  ~/rooks/rook-1.12.10/deploy/examples  kubectl version                        
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.14", GitCommit:"a5967a3c4d0f33469b7e7798c9ee548f71455222", GitTreeState:"clean", BuildDate:"2023-09-13T09:12:09Z", GoVersion:"go1.20.8", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.14", GitCommit:"a5967a3c4d0f33469b7e7798c9ee548f71455222", GitTreeState:"clean", BuildDate:"2023-09-13T09:04:55Z", GoVersion:"go1.20.8", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):  community version
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

**Comment by travisn** [comment]
Do you have [mem/cpu limits](https://rook.io/docs/rook/latest/CRDs/Cluster/ceph-cluster-crd/#resource-requirementslimits) applied to the OSDs? The helm chart applies them by default, but not the manifest install. I'm not sure the cause of the admin port issue you're seeing though.

**Comment by cyxinda** [comment]
> Do you have [mem/cpu limits](https://rook.io/docs/rook/latest/CRDs/Cluster/ceph-cluster-crd/#resource-requirementslimits) applied to the OSDs? The helm chart applies them by default, but not the manifest install. I'm not sure the cause of the admin port issue you're seeing though.

The setting yaml files in kubernetes is below:
[https://github.com/cyxinda/tmp/blob/main/rooks/rook-ceph-osd-31-8668454f9b-rfzgb.yaml](https://github.com/cyxinda/tmp/blob/main/rooks/rook-ceph-osd-31-8668454f9b-rfzgb.yaml)
I have setted the memory limits for osd pod,but not cpu limits.

**Comment by travisn** [comment]
Please open a [Ceph tracker](https://tracker.ceph.com/) for the admin socket issue.

**Comment by cyxinda** [comment]
> Please open a [Ceph tracker](https://tracker.ceph.com/) for the admin socket issue.

I have added a ceph tracker :[https://tracker.ceph.com/issues/64997]( https://tracker.ceph.com/issues/64997)
Thanks for paying attention to this issue.

**Comment by cyxinda** [comment]
this problem has been resolved in the version 1.13+ of rook.

**Comment by samcannell** [comment]
> this problem has been resolved in the version 1.13+ of rook.

I don't think that's correct -- I'm seeing it on a brand new cluster running Rook 1.15.5 and Ceph 19.2.0.  Symptoms are as described above -- the admin_socket process in one osd pod is always pegged at ~100% of one cpu.  If the affected OSD exits or is killed, another will take its place.

```
    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
3809010 167       20   0  875288 145100    856 R  99.7   0.1   6:12.80 admin_socket
```

**Comment by cyxinda** [comment]
> > this problem has been resolved in the version 1.13+ of rook.
> 
> I don't think that's correct -- I'm seeing it on a brand new cluster running Rook 1.15.5 and Ceph 19.2.0. Symptoms are as described above -- the admin_socket process in one osd pod is always pegged at ~100% of one cpu. If the affected OSD exits or is killed, a different OSD will start consuming high CPU.
> 
> ```
>     PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
> 3809010 167       20   0  875288 145100    856 R  99.7   0.1   6:12.80 admin_socket
> ```
I have encountered this problem in version 1.12.10.
When I  upgraded to rook-1.13.9 and the problem disappared.
![截图 2024-11-22 10-46-22](https://github.com/user-attachments/assets/1bfcefde-52ba-4436-99ea-41651b781a19)
Strictly, I have not tried Rook versions above 1.13.9 indeed.But you can try to change the parameter in containerd and see the effect.
[https://github.com/rook/rook/issues/12877#issuecomment-1905476914](https://github.com/rook/rook/issues/12877#issuecomment-1905476914)
